test/xUnit/csharp/test_CompletionHelpers.cs

```diff
@@ -92,5 +109,75 @@ public void TestHandleDoubleAndSingleQuote(string wordToComplete, string expecte
             Assert.Equal(expectedQuote, quote);
             Assert.Equal(expectedWordToComplete, wordToComplete);
         }
+
+        [Theory]
+        [InlineData("word", "word", true)]
+        [InlineData("Word", "word", true)]
+        [InlineData("word", "wor", true)]
+        [InlineData("word", "words", false)]
+        [InlineData("word`nnext", "word`n", true)]
+        [InlineData("word`r`nnext", "word`r`n", true)]
+        [InlineData("word;next", "word;", true)]
+        [InlineData("word,next", "word,", true)]
+        [InlineData("word[*]next", "word[*", true)]
+        [InlineData("word[abc]next", "word[abc", true)]
+        [InlineData("word", "word*", true)]
+        [InlineData("Word", "word*", true)]
+        [InlineData("word(Special)", "word*", true)]
+        [InlineData("testword", "test", true)]
+        [InlineData("word", "", true)]
+        [InlineData("", "word", false)]
+        public void TestDefaultMatch(string value, string wordToComplete, bool expected)
+        {
+            bool result = CompletionHelpers.DefaultMatch(value, wordToComplete);
+            Assert.Equal(expected, result);
+        }
+
+        [Theory]
+        [InlineData("word", "word", true)]
+        [InlineData("Word", "word", true)]
+        [InlineData("word", "wor", true)]
+        [InlineData("word", "words", false)]
+        [InlineData("word`nnext", "word`n", true)]
+        [InlineData("word`r`nnext", "word`r`n", true)]
+        [InlineData("word;next", "word;", true)]
+        [InlineData("word,next", "word,", true)]
+        [InlineData("word[*]next", "word[*", true)]
```

Comment:
This will cover second part of WildcardPatternEscapeMatch() method:
```suggestion
        [InlineData("word[*]next", "word[*", true)]
        [InlineData("word`?`[`*`]next", "word?[*", true)]
```